cni/pkg/ambient/net_linux.go

```diff
@@ -237,6 +237,49 @@ func getVethWithDestinationOf(ip string) (*netlink.Veth, error) {
 	return veth, nil
 }
 
+func findVethLinkForPeerIP(peerIP net.IP) (*netlink.Veth, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list links: %v", err)
+	}
+	var v netlink.Veth
+	for _, l := range links {
+		if l.Type() != v.Type() {
+			continue
+		}
+		veth := l.(*netlink.Veth)
+		peerIndex, err := getPeerIndex(veth)
+		if err != nil {
+			log.Warnf("failed to get peer index for veth interface %s: %v", veth.Name, err)
+		}
+		peerNs, err := getNsNameFromNsID(veth.Attrs().NetNsID)
+		if err != nil {
```

Comment:
> Why are we warning

I added these warning for debugging and then forgot to remove them.

> We should error out (or continue)

I think we should continue, because we don't know what interfaces exist on the host and actually we don't have to care about their correctness, so continuing makes it possible to find wanted interface on position n, while an interface on position n - 1 might return an error when getting net ns id or peer index.

Reply:
Ah yep, missed that. Thanks!